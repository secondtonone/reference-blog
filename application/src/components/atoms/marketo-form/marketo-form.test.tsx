/**
 * @jest-environment jsdom
 */

import { render } from '@testing-library/react';

import fixture from '~/__fixtures__/marketo-form';

import MarketoForm, { IMarketoProps } from '.';

const marketoFormProps = {
  ...fixture
} as IMarketoProps;

describe('Match snapshots MarketoForm', () => {
  let props: IMarketoProps;

  beforeEach(() => {
    props = { ...marketoFormProps };
  });

  it('default', () => {
    const { asFragment } = render(<MarketoForm {...props} />, {});
    expect(asFragment()).toMatchSnapshot();
  });

  it('without baseUrl', () => {
    delete props.baseUrl;
    const { asFragment } = render(<MarketoForm {...props} />, {});
    expect(asFragment()).toMatchSnapshot();
  });

  it('without fromId', () => {
    delete props.formId;
    const { asFragment } = render((
      <MarketoForm {... props} />
    ), {});
    expect(asFragment()).toMatchSnapshot();
  });

  it('without munchkinId', () => {
    delete props.munchkinId;
    const { asFragment } = render((
      <MarketoForm {...props} />
    ), {});
    expect(asFragment()).toMatchSnapshot();
  });
});
