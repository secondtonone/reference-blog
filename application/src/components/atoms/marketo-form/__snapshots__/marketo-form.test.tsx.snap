// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match snapshots MarketoForm default 1`] = `
<DocumentFragment>
  <form
    data-test="marketo-form"
    id="mktoForm_1073"
    style="display: none;"
  />
</DocumentFragment>
`;

exports[`Match snapshots MarketoForm without baseUrl 1`] = `
<DocumentFragment>
  <form
    data-test="marketo-form"
    id="mktoForm_1073"
    style="display: none;"
  />
</DocumentFragment>
`;

exports[`Match snapshots MarketoForm without fromId 1`] = `
<DocumentFragment>
  <form
    data-test="marketo-form"
    id="mktoForm_"
    style="display: none;"
  />
</DocumentFragment>
`;

exports[`Match snapshots MarketoForm without munchkinId 1`] = `
<DocumentFragment>
  <form
    data-test="marketo-form"
    id="mktoForm_1073"
    style="display: none;"
  />
</DocumentFragment>
`;
